sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/InvalidCastToInterface.cs

```diff
@@ -156,6 +156,7 @@ private ProgramState ProcessCastAccess(ProgramState programState, CastExpression
                 var type = semanticModel.GetTypeInfo(castExpression.Type).Type;
 
                 if (type == null ||
+                    type.OriginalDefinition.Is(KnownType.System_Nullable_T) ||
                     !semanticModel.Compilation.ClassifyConversion(typeExpression, type).IsNullable ||

```

Comment:
I could not find what ClassifyConversion does, but isn't it supposed to cover the same check as the one you added? Perhaps I am wrong since it did not work before...